module/zfs/zio_compress.c

```diff
@@ -126,9 +122,9 @@ zio_compress_zeroed_cb(void *data, size_t len, void *private)
 
 size_t
 zio_compress_data(enum zio_compress c, abd_t *src, void **dst, size_t s_len,
-    uint8_t level)
+    uint8_t level, size_t max_size)
```

Comment:
The list of arguments looks messy to me.  I would call it d_len and put after s_len.

Reply:
Great to have fresh eyes on it, d_len makes it consistent with other *(de)compress_data() functions, done.